File: Day1/src/domain/Elf.java
package domain;

import java.util.ArrayList;
import java.util.List;

public class Elf {
    private final List<Integer> foodCalories = new ArrayList<>();

    public void addFood(int calories) {
        foodCalories.add(calories);
    }

    public boolean hasFood() {
        return !foodCalories.isEmpty();
    }

    public int getCalories() {
        return foodCalories.stream().reduce(Integer::sum).orElse(0);
    }
}

File: Day1/src/domain/Elf.kt
package domain

class Elf {
    private val foodCalories = mutableListOf<Int>()
    fun addFood(calories: Int) {
        foodCalories.add(calories)
    }

    fun hasFood(): Boolean = foodCalories.isNotEmpty()

    fun calories(): Int = foodCalories.stream().reduce { a: Int, b: Int -> Integer.sum(a, b) }.orElse(0)
}